.github/workflows/release.yaml

```diff
@@ -8,15 +8,23 @@ on:
 jobs:
   publish:
     runs-on: ubuntu-latest
+    
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.RELEASE_PAT2 }}
```

Comment:
@ymansurozer I was going to use the same logic we used in grape-common and had prepared a comment for you, asking for a new PAT with rights to bypass branch protection. However, I want to give this suggestion a try, as it says `Since this is a push to the main branch, the default GITHUB_TOKEN should have sufficient permissions for this operation`. Let's make a call after your review about it, check if this works, and decide whether we do the same for grape-common (as well as other repositories we use changelogen)